File: scripts/run-tool.js
#!/usr/bin/env node

// Tool runner for JavaScript tools
// Usage: run-tool.js <tool_name> <json_args>

const fs = require('fs');
const path = require('path');

if (process.argv.length !== 4) {
    console.error('Usage: node run-tool.js <tool_name> <json_args>');
    process.exit(1);
}

const toolName = process.argv[2];
const jsonArgs = process.argv[3];

// Find the tool script
const toolScript = path.join('tools', `${toolName}.js`);
if (!fs.existsSync(toolScript)) {
    console.error(`Tool not found: ${toolScript}`);
    process.exit(1);
}

try {
    // Parse JSON arguments
    const args = JSON.parse(jsonArgs);
    
    // Load and execute the tool
    const tool = require(path.resolve(toolScript));
    if (typeof tool.run === 'function') {
        tool.run(args);
    } else {
        console.error(`Tool ${toolName} does not export a run function`);
        process.exit(1);
    }
} catch (error) {
    console.error(`Error executing tool ${toolName}:`, error.message);
    process.exit(1);
}

File: scripts/run-tool.py
#!/usr/bin/env python3

# Tool runner for Python tools
# Usage: run-tool.py <tool_name> <json_args>

import sys
import json
import importlib.util
import os

if len(sys.argv) != 3:
    print('Usage: python run-tool.py <tool_name> <json_args>', file=sys.stderr)
    sys.exit(1)

tool_name = sys.argv[1]
json_args = sys.argv[2]

# Find the tool script
tool_script = os.path.join('tools', f'{tool_name}.py')
if not os.path.exists(tool_script):
    print(f'Tool not found: {tool_script}', file=sys.stderr)
    sys.exit(1)

try:
    # Parse JSON arguments
    args = json.loads(json_args)
    
    # Load the tool module
    spec = importlib.util.spec_from_file_location(tool_name, tool_script)
    tool_module = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(tool_module)
    
    # Execute the tool
    if hasattr(tool_module, 'run'):
        tool_module.run(**args)
    else:
        print(f'Tool {tool_name} does not have a run function', file=sys.stderr)
        sys.exit(1)
        
except Exception as error:
    print(f'Error executing tool {tool_name}: {error}', file=sys.stderr)
    sys.exit(1)